src/v/storage/mvlog/entry_stream_utils.cc

```diff
@@ -0,0 +1,47 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "storage/mvlog/entry_stream_utils.h"
+
+#include "hashing/crc32.h"
+
+namespace storage::experimental::mvlog {
+
+iobuf entry_header_to_iobuf(const entry_header& hdr) {
+    iobuf b;
+    serde::write(b, hdr.header_crc);
+    serde::write(b, hdr.body_size);
+
+    // NOTE: serde serializes enums as int32_t (serde_enum_serialized_t).
+    // Explicitly serialize as int8_t.
+    serde::write(b, static_cast<int8_t>(hdr.type));
```

Comment:
this is what we expected users to do if they really didn't want to have a full sized integer for enum in serde. could optimize later, but the flip side is that we also probably don't need an int32_t for entry size and could take a byte from that. anyway, lgtm.

Reply:
Yeah this seemed like the "blessed" way we've handled them so far. I didn't want to get too fancy in optimizing for space, but we can revisit before the storage format is finalized if the additional space ends up being problematic